docker/dev/Dockerfile

```diff
@@ -1,35 +1,86 @@
-# syntax=docker/dockerfile:experimental
-FROM rust:1.47-buster as rust
+ARG RUST_BUILDER_IMAGE=ekidd/rust-musl-builder:stable
 
-ENV HOME=/home/root
+# Cargo chef plan
+FROM $RUST_BUILDER_IMAGE as planner
+WORKDIR /app
+
+# Copy dirs
+COPY Cargo.toml Cargo.lock ./
+COPY ./src ./src
+COPY ./migrations ./migrations
+COPY ./lemmy_api ./lemmy_api
+COPY ./lemmy_apub ./lemmy_apub
+COPY ./lemmy_db ./lemmy_db
+COPY ./lemmy_rate_limit ./lemmy_rate_limit
+COPY ./lemmy_structs ./lemmy_structs
+COPY ./lemmy_utils ./lemmy_utils
+COPY ./lemmy_websocket ./lemmy_websocket
```

Comment:
Why dont you just `COPY . .` here? Same in the block below, they look identical.

Reply:
We have the volumes folders in the `.dockerignore`, but even config file changes, or ansible template changes for example, or changes to scripts, will re-do the copy command. Fortunately it doesn't rebuild the project in that case, but that extra copying context takes ~ 30 seconds for me. This also can take advantage of specific cached source file folders. Basically I found its much better to manually copy the cargo files and source code files, than the whole dir.

Reply:
Then why not just add everything except source code, cargo files and docs to .dockerignore?

Reply:
The docs, and config.hjson can't be added to the dockerignore, since they are used by the runner and the docs builder.

Reply:
Okay then leave those out of the dockerignore, it doesnt matter if they are copied in to the builder. As long as `ansible/`, `docker/` and `*.sh` are ignored, there should be no problem doing `COPY . .`. Or maybe `COPY lemmy_* .` would work.